_posts/2022-02-28-python-shorts-unit-testing.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Python Shorts: The Benefits of Unit Testing"
+author:
+  display_name: Jake Adams
+  email: jdadams@utah.gov
+date: 2021-02-28 11:00:00
+categories:
+  - Featured
+  - Developer
+tags:
+  - python
+---
+
+In the realm of programming, "unit testing" is an automated way of testing small, individual "units" of your code to ensure that when given a specific set of inputs they return a specific set of outputs in the way you expected them to.
```

Comment:
**thought**: the same inputs same outputs thing is called a [pure function](https://en.wikipedia.org/wiki/Pure_function) and also can relate to [idempotence](https://en.wikipedia.org/wiki/Idempotence)

Reply:
Interesting, thanks for the references. 

If I'm understanding it correctly I don't think I mean to refer to completely pure functions; a test could make sure that an instance method sets a property on its object. My thought was that given a defined state and set of inputs, a function or method produces the expected outputs and side effects. That seems similar to pure functions but broader. Am I understanding that right?